distributed-commandbus-springcloud/src/main/java/org/axonframework/springcloud/commandhandling/SpringCloudHttpBackupCommandRouter.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2010-2017. Axon Framework
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.springcloud.commandhandling;
+
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.commandhandling.distributed.RoutingStrategy;
+import org.axonframework.commandhandling.distributed.SimpleMember;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.discovery.DiscoveryClient;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+import java.util.function.Predicate;
+
+/**
+ * Implementation of the {@link org.axonframework.springcloud.commandhandling.SpringCloudCommandRouter} which has a
+ * backup mechanism to request and provide a member and/or its {@link MessageRoutingInformation}.
+ * It uses {@link org.springframework.web.client.RestTemplate} to request {@code MembershipInformation} and services the
+ * purposes of a queryable location to retrieve the local {@code MembershipInformation} from by being a
+ * {@link org.springframework.web.bind.annotation.RestController}.
+ *
+ * @author Steven van Beelen
+ */
+@RestController
+@RequestMapping(SpringCloudHttpBackupCommandRouter.MESSAGE_ROUTING_INFORMATION_PATH)
+public class SpringCloudHttpBackupCommandRouter extends SpringCloudCommandRouter {
+
+    public static final String MESSAGE_ROUTING_INFORMATION_PATH = "/message-routing-information";
+    private static final Predicate<ServiceInstance> ACCEPT_ALL_INSTANCES_FILTER = serviceInstance -> true;
```

Comment:
Note that these predicates need to be serializable. Lambda expressions aren't. Instead, use an regular class instance that is safe to serialize and share between nodes.
There is a `AcceptAll` class exactly for that purpose

Reply:
This Predicate is to filter `ServiceInstances`,  it's not a `CommandMessage` filter like `AcceptAll` is.

Additionally, this predicate is only used by the `SpringCloudCommandRouter` and not propagated to the outside.
Do you still think I need to provide an implementation for it?

Reply:
You're right... please ignore my premature pavlov reaction ;-)